penrose-web/src/compiler/Style.ts

```diff
@@ -1763,31 +1772,37 @@ const genOptProblemAndState = (trans: Translation): State => {
 
 //#endregion
 
-// TODO: Improve this type signature
-// export const compileStyle = (env: VarEnv, subAST: SubProg, styAST: StyProg): State => {
-export const compileStyle = (stateJSON: any, styJSON: any): Either<StyErrors, State> => {
+export const parseStyle = (p: string): StyProg => {
+  const parser = new nearley.Parser(nearley.Grammar.fromCompiled(styleGrammar));
 
-  const info = stateJSON.default.contents;
-  console.log("compiling style (stateJSON)", info);
+  const { results } = parser.feed(p);
+  const ast: StyProg = results[0] as StyProg;
+  return ast;
+};
 
-  console.log("compiled style with new parser", styJSON.default);
+export const loadProgs = (): [Env, SubstanceEnv, SubProg, StyProg] => {
+  // TODO: Replace this with the sugared version...
+  // console.log("progs", domainStr, subStrUnsugared, styStr);
 
-  // Types from compileTrio
-  const state: State = info[0];
-  // const env: VarEnv = info[1]; // This is redundant with subOut
-  // const styProgInit: StyProg = info[2];
-  const styProgInit: StyProg = styJSON.default;
-  const subOut: SubOut = info[3];
+  const domainProgRes: Result<Env, PenroseError> = compileDomain(domainStr);
+  const env0: Env = unsafelyUnwrap(domainProgRes);
 
-  const subProg: SubProg = subOut[0];
-  const varEnv: VarEnv = subOut[1][0];
-  const subEnv: SubEnv = subOut[1][1];
-  // TODO: Bring back `eqEnv`?
-  const labelMap: LabelMap = subOut[2];
+  // TODO: Could be more efficient if compileSubstance also outputs parsed Sub program
```

Comment:
Didn't know `compileStyle` will be using the Substance AST. Will it be okay if I add the AST as a key in `SubstanceEnv`, instead of adding yet another member to the list of return values?

Reply:
Side note: you'll probably get to this when you are closer to the finish line. Remember to stop using `unsafelyUnwrap`. As the name suggests, it's a function for debugging. You can do an `if` check by calling `isOk()` on the result.

Reply:
> Didn't know compileStyle will be using the Substance AST. 

Yes, it's needed to match selectors against the Substance program.

> Will it be okay if I add the AST as a key in SubstanceEnv, instead of adding yet another member to the list of return values?

Yes, that's fine.

> Remember to stop using unsafelyUnwrap. As the name suggests, it's a function for debugging. You can do an if check by calling isOk() on the result or use functions like andThen. I can help you refactor this after you are done with other things.

It looks like this `Result` monad is going to require using it and its helper functions everywhere in the Style compiler, right? To test things, I just need the value out of the `Result`; the rest of it seems to be related to error handling so I'm going to leave this for later. It would be helpful if you can address the `Result` refactor when we get to error handling in the Style compiler.

Reply:
> It looks like this `Result` monad is going to require using it and its helper functions everywhere in the Style compiler, right? To test things, I just need the value out of the `Result`; the rest of it seems to be related to error handling so I'm going to leave this for later. It would be helpful if you can address the `Result` refactor when we get to error handling in the Style compiler.

It shouldn't be that viral in the Style compiler. My plan is to only use `Result` in `compileStyle` and all other style errors will be handled as usual, just a giant catch in `compileStyle`. We can do that once this PR is merged.

Reply:
Ok, that sounds good.